packages/ckeditor5-ui/src/search/text/searchtextview.ts

```diff
@@ -0,0 +1,414 @@
+			actions: {
+				// Navigate form fields backwards using the Shift + Tab keystroke.
+				focusPrevious: 'shift + tab',
+
+				// Navigate form fields forwards using the Tab key.
+				focusNext: 'tab'
+			}
+		} );
+
+		this.on<SearchTextViewSearchEvent>( 'search', ( evt, { resultsCount, totalItemsCount } ) => {
+			this.resultsCount = resultsCount;
+			this.totalItemsCount = totalItemsCount;
+		} );
+
+		this.setTemplate( {
+			tag: 'div',
+			attributes: {
+				class: [
+					'ck',
+					'ck-search',
+					config.class || null
+				],
+
+				tabindex: '-1'
+			},
+			children: this.children
+		} );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override render(): void {
+		super.render();
+
+		this.children.addMany( [
+			this.queryView,
+			this.resultsView
+		] );
+
+		const stopPropagation = ( data: Event ) => data.stopPropagation();
+
+		for ( const focusableChild of this.focusableChildren ) {
+			this.focusTracker.add( focusableChild.element as Element );
+		}
+
+		// Start listening for the keystrokes coming from #element.
+		this.keystrokes.listenTo( this.element as HTMLElement );
+
+		// Since the form is in the dropdown panel which is a child of the toolbar, the toolbar's
+		// keystroke handler would take over the key management in the URL input. We need to prevent
+		// this ASAP. Otherwise, the basic caret movement using the arrow keys will be impossible.
+		this.keystrokes.set( 'arrowright', stopPropagation );
+		this.keystrokes.set( 'arrowleft', stopPropagation );
+		this.keystrokes.set( 'arrowup', stopPropagation );
+		this.keystrokes.set( 'arrowdown', stopPropagation );
+	}
+
+	/**
+	 * Focuses the {@link #queryView}.
+	 */
+	public focus(): void {
+		this.queryView.focus();
+	}
+
+	/**
+	 * Resets the component to its initial state.
+	 */
+	public reset(): void {
+		this.queryView.reset();
+		this.search( '' );
+	}
+
+	/**
+	 * Searches the {@link #filteredView} for the given query.
+	 *
+	 * @internal
+	 * @param query The search query string.
+	 */
+	public search( query: string ): void {
+		const regExp = query ? new RegExp( escapeRegExp( query ), 'ig' ) : null;
+		const filteringResults = this.filteredView.filter( regExp );
+
+		this.fire<SearchTextViewSearchEvent>( 'search', { query, ...filteringResults } );
+	}
+
+	/**
+	 * Creates a search field view based on configured creator..
+	 */
+	private _createSearchTextQueryView(): SearchTextQueryView<TQueryFieldView> {
+		const queryView = new SearchTextQueryView<TQueryFieldView>( this.locale, this._config.queryView );
+
+		this.listenTo( queryView.fieldView, 'input', () => {
+			this.search( queryView.fieldView.element!.value );
+		} );
+
+		queryView.on( 'reset', () => this.reset() );
+		queryView.bind( 'isEnabled' ).to( this );
+
+		return queryView;
+	}
+
+	/**
+	 * Initializes the default {@link #infoView} behavior with default text labels when no custom info view
+	 * was specified in the view config.
+	 */
+	private _enableDefaultInfoViewBehavior(): void {
```

Comment:
As mentioned in comment below, I think that the component should not be searchable if it is empty. "Empty" and "no results" should be handled separately. But we can move this to a followup issue.

Reply:
Let's discuss this in a follow-up please.

Reply:
Would you please add an issue for that with very short description? Then we can resolve these comments.

Reply:
https://github.com/ckeditor/ckeditor5/issues/15059